packages/affiliates/libs/models.js

```diff
@@ -0,0 +1,184 @@
+const assert = require("assert");
+const lodash = require("lodash");
+
+// assumes sorted oldest to newest
+// always return price equal to or earlier than requested time
+// prices are input directly from coingecko api
+// [
+//   [ timestamp, price],
+//   [ timestamp, price],
+// ]
+function Prices(prices = []) {
+  function lookup(time) {
+    const index = lodash.sortedIndexBy(prices, [time], ([time]) => {
+      return time;
+    });
+    if (prices[index] === time) return prices[index];
+    return prices[index - 1];
+  }
+
+  return {
+    lookup,
+    prices
+  };
+}
+
+// users show current ratio of affiliates tagged on thier deposits
+// keyed by user address, and show total attributed for each affiliate per user
+function SharedAttributions() {
```

Comment:
Why not set scale in the constructor rather than passing it in on every method call?